AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreLocalUsageTests.swift

```diff
@@ -0,0 +1,289 @@
+                return
+            }
+            if rating < previousRating {
+                XCTFail("Retured posts are not in expected order")
+            } else {
+                previousRating = rating
+            }
+        }
+    }
+
+    func testTwoSorts() throws {
+        _ = saveData(num: 50)
+
+        let queryFinished = expectation(description: "Query post completed")
+        var posts = [Post]()
+
+        Amplify.DataStore.query(Post.self,
+                                sort: .by(.ascending(Post.keys.rating),
+                                          .descending(Post.keys.title))) { res in
+            switch res {
+            case .success(let returnPosts):
+                posts = returnPosts
+                queryFinished.fulfill()
+            case .failure(let error):
+                XCTFail("Error quering posts, \(error)")
+            }
+        }
+        wait(for: [queryFinished], timeout: 10)
+
+        var previousRating: Double = 0
+        for post in posts {
+            guard let rating = post.rating else {
+                XCTFail("Rating should not be nil")
+                return
+            }
+            if rating < previousRating {
+                XCTFail("Retured posts are not in expected order")
+            } else {
+                previousRating = rating
+            }
+        }
+
+        // https://developer.apple.com/documentation/swift/dictionary/3127163-init
+        let postsDic = Dictionary(grouping: posts, by: { $0.rating! })
+        for (_, pairs) in postsDic {
+            for index in 0 ..< pairs.count - 1 {
+                print(pairs[index].title)
+                if pairs[index].title < pairs[index + 1].title {
+                    XCTFail("Post titles are not in expected orders")
+                }
+            }
+        }
+    }
+
+    func testPredicateAndSort() throws {
+        _ = saveData(num: 20)
+
+        let queryFinished = expectation(description: "Query post completed")
+        var posts = [Post]()
+
+        Amplify.DataStore.query(Post.self,
+                                where: Post.keys.rating >= 2,
+                                sort: .ascending(Post.keys.rating)) { res in
+            switch res {
+            case .success(let returnPosts):
+                posts = returnPosts
+                queryFinished.fulfill()
+            case .failure(let error):
+                XCTFail("Error quering posts, \(error)")
+            }
+        }
+        wait(for: [queryFinished], timeout: 10)
+
+        var previousRating: Double = 0
+        for post in posts {
+            guard let rating = post.rating else {
+                XCTFail("Rating should not be nil")
+                return
+            }
+            if rating < previousRating {
+                XCTFail("Retured posts are not in expected order")
+            } else if rating < 2 {
+                XCTFail("Predicate is not working as expected")
```

Comment:
this can be pulled out of the `else if` and be it's own if statement so the logic becomes 
- make sure it's in ascending order
- make sure the rating value has respected the where parameter

Reply:
Good point, changed